main.js

```diff
@@ -1,352 +1 @@
-const pvpForm = document.getElementById('pvpForm');
-const pvcForm = document.getElementById('pvcForm');
-const winDiv = document.getElementById('winDiv');
-const tieDiv = document.getElementById('tieDiv');
-
-
-// Opens 2-Player form
-function toggleModal(e) {
-    e.preventDefault() // Prevent form submission
-    const id = e.target.getAttribute('id'); // get id
-
-    // Hide modal
-    if (id === 'modal-shadow' || id === 'modal2' || id === 'modal1') {
-        shadow.style.display = 'none';
-        modal.style.display = 'none';
-        pvpForm.style.display = 'none';
-        pvcForm.style.display = 'none';
-        winDiv.style.display = 'none';
-        tieDiv.style.display = 'none';
-        return;
-    }
-    // Display modal
-    shadow.style.display = 'block';
-    if (id === 'pvp') {
-        // Display 2 player form
-        modal.style.display = 'flex';
-        pvpForm.style.display = 'grid';
-        return;
-    }
-    // Display 1 player form
-    modal.style.display = 'flex';
-    pvcForm.style.display = 'grid'
-}
-
-// Open modals on button click
-const pvpButton = document.getElementById('pvp'); // 2-Player
-pvpButton.addEventListener('click', toggleModal);
-const pvcButton = document.getElementById('pvc'); // 1-Player
-pvcButton.addEventListener('click', toggleModal);
-
-// Close modal on shadow click
-shadow.addEventListener('click', toggleModal)
-
-// An object that stores the game board & its actions
-const gameBoard = {
-    tiles: [   
-        document.getElementById('t0'),
-        document.getElementById('t1'),
-        document.getElementById('t2'),
-        document.getElementById('t3'),
-        document.getElementById('t4'),
-        document.getElementById('t5'),
-        document.getElementById('t6'),
-        document.getElementById('t7'),
-        document.getElementById('t8')
-    ],
-    getIcons(index) {
-        const ghost = document.getElementById('ghost' + index);
-        const skull = document.getElementById('skull' + index);
-        return {ghost, skull};
-    },
-    showIcon(index, string) {
-        const icons = this.getIcons(index);
-        if (string === 'skull'){
-            icons.skull.style.display = 'block'; 
-            return; 
-        }
-        icons.ghost.style.display = 'block'; 
-        return; 
-    }, 
-    clearIcons() { 
-        for (let i = 0; i < 9; i++){
-            let icons = this.getIcons(i); 
-            icons.skull.style.display = 'none'; 
-            icons.ghost.style.display = 'none'; 
-        }
-        return; 
-    },
-    isAvailable(index) { 
-        console.log('entered isAvailable: ', index);
-        return this.tiles[index].getAttribute('class') === 'gameTile';
-    },
-    availableTiles() { // return an array of all available tiles
-        let availableTiles = [];
-        for (let i = 0; i < 9; i++){
-            if (this.isAvailable(i)){
-                availableTiles.push(i);
-            }
-        }
```

Comment:
this is fine but the Array.filter method is finer

this.tiles.filter(tile => tile.getAttribute('class') === 'gameTile')